src/java.base/share/classes/sun/security/jca/ProviderList.java

```diff
@@ -655,27 +656,27 @@ public String toString() {
     }
 
     /* Defined Groups for jdk.security.provider.preferred */
-    private static final String SHA2Group[] = { "SHA-224", "SHA-256",
+    private static final String[] SHA2Group = { "SHA-224", "SHA-256",
             "SHA-384", "SHA-512", "SHA-512/224", "SHA-512/256" };
-    private static final String HmacSHA2Group[] = { "HmacSHA224",
+    private static final String[] HmacSHA2Group = { "HmacSHA224",
             "HmacSHA256", "HmacSHA384", "HmacSHA512"};
-    private static final String SHA2RSAGroup[] = { "SHA224withRSA",
+    private static final String[] SHA2RSAGroup = { "SHA224withRSA",
             "SHA256withRSA", "SHA384withRSA", "SHA512withRSA"};
-    private static final String SHA2DSAGroup[] = { "SHA224withDSA",
+    private static final String[] SHA2DSAGroup = { "SHA224withDSA",
             "SHA256withDSA", "SHA384withDSA", "SHA512withDSA"};
-    private static final String SHA2ECDSAGroup[] = { "SHA224withECDSA",
+    private static final String[] SHA2ECDSAGroup = { "SHA224withECDSA",
             "SHA256withECDSA", "SHA384withECDSA", "SHA512withECDSA"};
-    private static final String SHA3Group[] = { "SHA3-224", "SHA3-256",
+    private static final String[] SHA3Group = { "SHA3-224", "SHA3-256",
             "SHA3-384", "SHA3-512" };
-    private static final String HmacSHA3Group[] = { "HmacSHA3-224",
+    private static final String[] HmacSHA3Group = { "HmacSHA3-224",
             "HmacSHA3-256", "HmacSHA3-384", "HmacSHA3-512"};
 
     // Individual preferred property entry from jdk.security.provider.preferred
     private static class PreferredEntry {
         private String type = null;
-        private String algorithm;
-        private String provider;
-        private String alternateNames[] = null;
+        private final String algorithm;
+        private final String provider;
+        private String[] alternateNames = null;
```

Comment:
shouldn't this also be final?

Reply:
If I make alternateNames final, then IJ complains "cannot assign a value to final variable".
If I make  it final and remove initialization, then make complains that alternateNames might not have been initialized.

Reply:
I would make it final, and then assign it to null in a final else block after line 719.

Reply:
Fixed. I was also able to make `type` final on line 676.